plugins/guests/windows/cap/reboot.rb

```diff
@@ -1,21 +1,51 @@
+require "log4r"
+
 module VagrantPlugins
   module GuestWindows
     module Cap
       class Reboot
-        def self.wait_for_reboot(machine)
-          # Technically it should be possible to make it work with SSH
-          # too, but we don't yet.
-          return if machine.config.vm.communicator != :winrm
+        MAX_REBOOT_RETRY_DURATION = 120
+
+        def self.reboot(machine)
+          @logger = Log4r::Logger.new("vagrant::windows::reboot")
+          reboot_script = "shutdown /r /t 5 /f /d p:4:1 /c \"Vagrant Reboot Computer\""
+
+          comm = machine.communicate
+
+          script  = File.expand_path("../../scripts/reboot_detect.ps1", __FILE__)
+          script  = File.read(script)
+          if comm.test(script, error_check: false, shell: :powershell)
+            @logger.debug("Issuing reboot command for guest")
+            comm.execute(reboot_script, shell: :powershell)
+          else
+            @logger.debug("A reboot is already in progress")
+          end
+
+          @logger.debug("Waiting for machine to finish rebooting")
 
+          wait_remaining = MAX_REBOOT_RETRY_DURATION
+          begin
+            wait_for_reboot(machine)
+          rescue Vagrant::Errors::MachineGuestNotReady, WinRM::WinRMHTTPTransportError => e
```

Comment:
Only the `Vagrant::Errors::MachineGuestNotReady` exception should be needed here. Also the `=> e` can be dropped since it is unused.